cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorTest.java

```diff
@@ -125,7 +128,153 @@ public void testAllTypesEvaluationSuccess() throws Exception {
           "-o", "B=" + bFile.toURI().toString(),
           "-o", "C=" + cFile.toURI().toString(),
           "-o", "D=" + dFile.toURI().toString(),
-          "-n", "10"
+          "-n", "10",
+          "--overwrite-output-for-contexts"
+        };
+
+        SparkCqlEvaluator.main(args);
+
+        // Expected rows per context were derived from inspecting the input data
+        // by hand and counting the unique values for each context's key column.
+        validateOutputCountsAndColumns(patientFile.toURI().toString(), new HashSet<>(Arrays.asList("pat_id", "MeasureAB|cohort")), 100);
+        validateOutputCountsAndColumns(aFile.toURI().toString(), new HashSet<>(Arrays.asList("id_col", "MeasureA|cohort")), 572);
+        validateOutputCountsAndColumns(bFile.toURI().toString(), new HashSet<>(Arrays.asList("id", "MeasureB|cohort")), 575);
+        validateOutputCountsAndColumns(cFile.toURI().toString(), new HashSet<>(Arrays.asList("id", "MeasureC|cohort")), 600);
+        validateOutputCountsAndColumns(dFile.toURI().toString(), new HashSet<>(Arrays.asList("id", "MeasureD|cohort")), 567);
+    }
+
+    private void validateOutputCountsAndColumns(String filename, Set<String> columnNames, int numExpectedRows) {
+        // SparkCqlEvaluator closes the SparkSession. Make sure we have one opened before any validation. 
+        try(SparkSession sparkSession = initializeSession(Java8API.ENABLED)) {
```

Comment:
Unless there's a deeper reason, I say we just let the `SparkSession` objects live.
`SparkSession.Builder` should take care of reusing sessions for us and the more sessions we create, the slower our overall test suite is.

Reply:
Talked with Matt outside of this PR.  I don't think any changes are necessary here.